drivers/soundwire/cadence_master.c

```diff
@@ -17,6 +17,7 @@
 #include <linux/soundwire/sdw.h>
```

Comment:
fix commit message

soundwire: intel/cadence: fix timeouts in MSI mode

The existing code uses one pair of interrupt handler/thread per link, but at the hardware level the interrupt is shared. This works fine for legacy PCI interrupts, but leads to timeouts in MSI (Message-Signaled Interrupt) mode, likely due to edges being lost.

This patch refactors the interrupt handling with a single pair of handler/thread for all links. The handler is simplified to the bare minimum of detecting a SoundWire interrupt, and the thread takes care of dealing with interrupt sources. This partition follows the model used for the SOF IPC on HDaudio platforms, where similar timeout issues were noticed and doing all the interrupt handling/clearing in the thread improved reliability/stability.

Validation results with 4 links active in parallel show a night-and-day improvement with no timeouts noticed even during stress tests. Latency and quality of service are not affected by the change - mostly because events on a SoundWire link are throttled by the bus frame rate (typically 8..48kHz).

Reply:
> This patch refactors the interrupt handling with a single pair of handler/thread for all links.

@plbossart this is still confusing. It might be better to say that "This patch unifies interrupt handling for all links with a single threaded IRQ handler"

Reply:
+1